search_service/proxy/atlas.py

```diff
@@ -172,82 +286,62 @@ def fetch_table_search_results_with_field(self, *,
+                                               index: str = '') -> SearchResult:
         """
-        :param query_params: A dictionary of query parameter need to pass
-        to Basic Search Post method of Atlas.
-        :return: list of tables, along with the approximate count
+        Conduct an 'Advanced Search' to narrow down search results with a use of filters.
+
+        Using Atlas Basic Search with filters to retrieve precise results and sort them by popularity score.
+
+
+        :param query_term: A Search Query Term
+        :param search_request: Values from Filters
+        :param page_index: Index of search page user is currently on (for pagination)
+        :param index: Search Index (different resource corresponding to different index)
+        :return: SearchResult Object
         """
-        try:
-            # Fetch the table entities based on query terms
-            table_results = self.atlas.search_basic.create(data=query_params)
-        except BadRequest as ex:
-            LOGGER.error(f"Fetching Tables Failed : {str(ex)}")
-            return [], 0
+        _filters = search_request.get('filters', dict())
 
-        if not len(table_results.entities):
-            return [], 0
+        db_filter_value = _filters.get('database')
+        table_filter_value = _filters.get('table')
+        cluster_filter_value = _filters.get('cluster')
+        badges_filter_value = _filters.get('badges', list())
+        tags_filter_value = _filters.get('tag', list())
 
-        # noinspection PyProtectedMember
-        tables_count = table_results._data.get("approximateCount")
-
-        tables = []
-        for table in table_results.entities:
-            table_attrs = table.attributes
-
-            table_qn = parse_table_qualified_name(
-                qualified_name=table_attrs.get(self.QN_KEY)
-            )
-
-            table_name = table_qn.get("table_name") or table_attrs.get('name')
-            db_name = table_qn.get("db_name", '')
-            db_cluster = table_qn.get("cluster_name", '')
-            table = Table(name=table_name,
-                          key=f"{table.typeName}://{db_cluster}.{db_name}/{table_name}",
-                          description=table_attrs.get('description') or table_attrs.get('comment'),
-                          cluster=db_cluster,
-                          database=table.typeName,
-                          schema=db_name,
-                          column_names=[],
-                          tags=[],
-                          badges=[],
-                          last_updated_timestamp=table_attrs.get('updateTime'))
+        filters = list()
 
-            tables.append(table)
+        if db_filter_value:
+            filters.append((self.ATLAS_QN_ATTRIBUTE, 'STARTSWITH', db_filter_value[0] + '.'))
 
-        return tables, tables_count
+        if cluster_filter_value:
+            filters.append((self.ATLAS_QN_ATTRIBUTE, 'ENDSWITH', '@' + cluster_filter_value[0]))
```

Comment:
note to self: Provide a proper typed Python object that exposes various parts of a QN as attributes.